### PR TITLE
Send notifications when entity created, updated or invited

### PR DESCRIPTION
**Pull request by varadeth**
Configure a way to send notifications when a entity is created, updated or invited either by using kafka or through actor which call rest api to send the notification object to rest api. The notification template will be picked from schema. Default template is already configured

Resolves [#656](https://github.com/Sunbird-RC/community/issues/626)

### CODE REVIEW

File: java/registry/src/main/java/dev/sunbirdrc/registry/service/NotificationHelper.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        objectNodeMap.put(entityType, entityName);
+        return objectNodeMap;
+    }
+
+    private String getMessageString(String messageTemplateBody, Map<String, Object> objectNodeMap, String operationType) throws IOException {
+        Handlebars handlebars = new Handlebars();
+        Template messageTemplate = handlebars.compileInline(messageTemplateBody);
+        return messageTemplate.apply(objectNodeMap);
+    }
+    private void sendNotificationToOwners(JsonNode inputJson, String operation, String subject, String message) throws Exception {
```
[the remark this answers is not in the document]
Comment (varadeth) [reply java/registry/src/main/java/dev/sunbirdrc/registry/service/NotificationHelper.java:97]: Sending List<ObjectNode> itself. don't need that whole inputJson object

File: java/registry/src/main/java/dev/sunbirdrc/registry/helper/RegistryHelper.java [hunk 2]
```diff
@@ -383,6 +365,7 @@ public void updateEntityAndState(JsonNode existingNode, JsonNode updatedNode, St
             entityStateHelper.applyWorkflowTransitions(existingNode, updatedNode, attestationPolicies);
         }
         updateEntity(updatedNode, userId);
+        notificationHelper.sendNotification(updatedNode, UPDATE);
```
[the remark this answers is not in the document]
Comment (varadeth) [reply java/registry/src/main/java/dev/sunbirdrc/registry/helper/RegistryHelper.java:370]: added for delete and also for update property which was missing earlier

File: java/registry/src/test/java/dev/sunbirdrc/registry/helper/RegistryHelperTest.java [hunk 3]
```diff
@@ -893,8 +893,26 @@ public void shouldNotRaiseRequiredExceptionsIFFlagDisabled() throws Exception {
...
+		mockDefinitionManager();
+		mockValidationService();
+		when(shardManager.getShard(any())).thenReturn(new Shard());
+		when(dbConnectionInfoMgr.getUuidPropertyName()).thenReturn("osid");
+		ReflectionTestUtils.setField(registryHelper, "notificationEnabled", true);
+		doNothing().when(registryService).updateEntity(any(), any(), any(), any());
+		doNothing().when(notificationHelper).sendNotification(any(), any());
+		registryHelper.updateEntityAndState(existingJson, updateJson, "");
+		verify(registryService, times(1)).updateEntity(any(), any(), any(), any());
+		verify(notificationHelper, times(1)).sendNotification(any(), any());
```
[the remark this answers is not in the document]
Comment (varadeth) [reply java/registry/src/test/java/dev/sunbirdrc/registry/helper/RegistryHelperTest.java:917]: yes. i have added new tests in notificationhelper tests now

File: java/registry/src/main/java/dev/sunbirdrc/registry/model/NotificationTemplates.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        template = new NotificationTemplate("Credential Updated", "{\"sender\": \"AppName\",\"route\": \"4\",\"country\": \"91\",\"unicode\": 1,\"sms\": {    \"message\": \"{{name}}, Your {{entityType}} credential has been updated\",    \"to\": [      \"{{contact}}\"    ]  }],\"DLT_TE_ID\": \"templateId\"}");
+        updateNotificationTemplates = new ArrayList();
+        updateNotificationTemplates.add(template);
+
+        template = new NotificationTemplate("Invitation", "{\"sender\": \"AppName\",\"route\": \"4\",\"country\": \"91\",\"unicode\": 1,\"sms\": [  {    \"message\": \"{{name}}, You have been invited for {{entityType}}\",    \"to\": [      \"{{contact}}\"    ]  }],\"DLT_TE_ID\": \"templateId\"}");
+        inviteNotificationTemplates = new ArrayList<>();
+        inviteNotificationTemplates.add(template);
+
+        template = new NotificationTemplate("Revoked", "{\"sender\": \"AppName\",\"route\": \"4\",\"country\": \"91\",\"unicode\": 1,\"sms\": [  {    \"message\": \"{{name}}, Your {{entityType}} credential has been revoked\",    \"to\": [      \"{{contact}}\"    ]  }],\"DLT_TE_ID\": \"templateId\"}");
```
Comment (tejash-jl) [inline java/registry/src/main/java/dev/sunbirdrc/registry/model/NotificationTemplates.java:29]: is this default values?

**Review by tejash-jl — APPROVED** [review]

**Comment by tejash-jl** [comment]
Add the e2e tests

**Pull request merged**